src/worker.cc

```diff
@@ -175,8 +175,8 @@ rapidjson::Document from_string(const std::string& json, const valhalla_exceptio
 
 void add_date_to_locations(Options& options,
                            google::protobuf::RepeatedPtrField<valhalla::Location>& locations) {
-  // otherwise we do what the person was asking for
-  if (options.has_date_time_case() && !locations.empty()) {
+  // /route needs special treatment
+  if (options.action() == Options::route && options.has_date_time_case() && !locations.empty()) {
     switch (options.date_time_type()) {
```

Comment:
`date_time.type` is only relevant for routing, all other services should default all locations to the global `date_time.value`